util-macros/src/lib.rs

```diff
@@ -0,0 +1,29 @@
+extern crate proc_macro;
+
+use std::iter::FromIterator;
+
+#[proc_macro_attribute]
+pub fn test(
+    _args: proc_macro::TokenStream,
+    mut ts: proc_macro::TokenStream,
+) -> proc_macro::TokenStream {
+    let ts_clone = ts.clone();
+    let func = syn::parse_macro_input!(ts_clone as syn::ItemFn);
+
+    let ident = &func.sig.ident;
+    let test_name = ident.to_string();
+    let test_case_ident = quote::format_ident!("TEST_CASE_{}", ident);
+    let result = quote::quote! {
+        #[test_case]
+        const #test_case_ident: ::mycelium_util::testing::TestCase = ::mycelium_util::testing::TestCase {
+            name: #test_name,
+            func: || ::mycelium_util::testing::TestResult::into_result(#ident()),
```

Comment:
I tried to make it such that our tests can return results, so we can recover from failure, rather than handling panicing assertions.

It'd be nice to have each test run in it's own clean QEMU instance, but I wasn't sure what the best way to do that was, given that we don't have a good mechanism to pass info in yet. One option would be to have the default behaviour just be to print out all available tests, and have the test runner then reset & re-call the kernel with each test in order.

It's also inconvenient with how it's currently written, as tests written outside of the `mycelium_kernel` module don't share the same test runner, so we'd need to effectively write an entire boot routine for each integration test, etc. that we want to write.

I don't have a great answer here, unfortunately :'-(